### ISSUE TITLE
Media types for CQL?

### ISSUE THREAD

**Issue by cportele** [opengeospatial/ogcapi-features#467]
Is there value in registering media types for the text and JSON flavors of CQL, e.g. `application/cql+json` `application/cql+text`? 

That is, are there use cases where CQL will be used as payload directly or will it typically be embedded in the payload, e.g. as part of a query (that includes other aspects like the properties to return, sorting, etc.)? So far, we have not planned to register media types, because of the assumption that filter expressions will rarely be used as separate messages, but usually be part of a request or response. Are there use cases where a CQL media type would be important?

**Comment by ghobona** [comment]
One use case could be retrieval of a Stored Query that is encoded in CQL from either OGC API - Features or from OGC API - Records.

**Comment by pvretano** [comment]
@ghobona the filter would be embedded in the query in this case and so we would need a MIME type for the query but not the filter.  
For example (and this is just an example off the top of my head) a query expression might look like this:
```
{
   "collections": ["/collections/myFeatureCollection"],
   "properties": ["featProp1", "featProp2", "featProp3"],
   "cqlFilter": { ... filter object ...},
   "sortBy": { ... sort expression ...}
}
```
You would POST this to some endpoint to create the stored query and the GET from the created query to get the results.

So, we would need a MIME type for this **whole** thing ... not just the filter.

**Comment by cportele** [comment]
Meeting 2021-06-21: So far, we do not have a use case where a "raw" filter is sent. @pvretano mentioned a potential use case in the PubSub context. He will look to information. If there is a use case, we should register media types for the encodings, if not we wouldn't register media types.

**Comment by pvretano** [comment]
Checked the PubSub specification and there are no instances that I see where a filter is a standalone thing.  The case I was thinking of was [creating a subscription](http://docs.opengeospatial.org/is/13-131r1/13-131r1.html#27).  
There is no RESTful API defined for PubSub but I envisioned that simply posting a filter to a subscription resource would create a subscription.  However, it seem more information is required and just a filter to create a subscription and so the filter would likely be embedded into a larger containing structure.  So, the long and short of it is that it is unlikely we need a dedicated MIME type for filter for the PubSub case.
